bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -666,6 +659,15 @@ public Bookie(ServerConfiguration conf, StatsLogger statsLogger)
         this.ledgerMonitor = new LedgerDirsMonitor(conf, diskChecker, ledgerDirsManager);
         try {
             this.ledgerMonitor.init();
+
+            // if the index and ledger are sharing directories we can use the same monitor,
```

Comment:
that change here is not correct.

if ledgerMonitor is throwing `NoWritableLedgerDirException` and transition to readonly mode, the idxMonitor will never be initialized.

Reply:
it would be nice to create an issue to track your thought of the improvement. one single disk monitor is great.

Reply:
#1655